before_install_linux.sh

```diff
@@ -16,8 +16,12 @@ if [ ! -d /usr/include/freetype ]; then
 fi
 
 if [ -v IMAGEMAGICK_VERSION ]; then
-  wget http://www.imagemagick.org/download/releases/ImageMagick-${IMAGEMAGICK_VERSION}.tar.xz
-  tar -xf ImageMagick-${IMAGEMAGICK_VERSION}.tar.xz
+  if wget https://github.com/ImageMagick/ImageMagick/archive/${IMAGEMAGICK_VERSION}.tar.gz; then
```

Comment:
One thought on this is that if for any reason that the github url is broken, this will *always* fall back to the `else` case. What if instead we do something like:

```sh
# not sure of the exact syntax...
if [ IMAGEMAGICK_VERSION == 6.7 || IMAGEMAGICK_VERSION == 6.8 ]; then
  # download from imagemagick.org
else
  # download from github
end
```

Reply:
I don't understand what you are trying to tell me. But if the GitHub download fails for some reason it will try to download from ImageMagick and that will most likely fail because the patch version has been "updated".

Reply:
Yeah, that's true, though it may be a factor of timing. Some more thoughts on this:

- This approach makes 2 remote requests for some versions of ImageMagick, 1 to Github and then a second to imagemagick.org. Not a big deal, but it slightly slows down the tests and we already know where we want to go for each version, so we only need to make 1 request. It's more of a problem on Travis where the connection can be unreliable. We've seen a lot of random failures due to issues requesting Linux repos while updating packages (which might be good for us to remove).
- If we have an explicit mention of the version either in the line or in a comment nearby, it'll be easier to find it and remove it once we've deprecated support for these versions.
- If it does cause an error in the Github portion and fall back to imagemagick.org, if that fails too, somebody might not fix it the way we intend. Instead of examining the Github url, or just re-running the build, they may end up deciding to bump the patch level instead, which is what we're trying to avoid needing to do. If we're explicit about which versions should go to imagemagick.org, we know that any Github failures are directly related to Github, but any imagemagick.org failures are likely to be patch level issues.

All of these are relatively minor issues, though, so if you really want to keep it this way I don't mind. I'm mostly trying to minimize the possibility of a failure and make sure it's the minimum number of steps to find the cause when one happens.

Reply:
Just pushed a patch that will check the version number and determine were to download the file from. And this actually made me find some mistakes in the script. Good catch!